habitat/tasks/rearrange/multi_task/composite_sensors.py

```diff
@@ -5,14 +5,52 @@
 # LICENSE file in the root directory of this source tree.
 
 
+from typing import List
+
+import numpy as np
+from gym import spaces
+
 from habitat.core.embodied_task import Measure
 from habitat.core.registry import registry
+from habitat.core.simulator import Sensor, SensorTypes
 from habitat.tasks.rearrange.rearrange_sensors import (
     EndEffectorToObjectDistance,
     ObjectToGoalDistance,
     RearrangeReward,
 )
-from habitat.tasks.rearrange.utils import rearrange_logger
+
+
+@registry.register_sensor
+class GlobalPredicatesSensor(Sensor):
+    def __init__(self, sim, config, *args, task, **kwargs):
+        self._task = task
+        self._sim = sim
+        self._predicates_list = None
+        super().__init__(config=config)
+
+    def _get_uuid(self, *args, **kwargs):
+        return "all_predicates"
+
+    def _get_sensor_type(self, *args, **kwargs):
+        return SensorTypes.TENSOR
+
+    def _get_observation_space(self, *args, config, **kwargs):
+        if self._predicates_list is None:
```

Comment:
Why the lazy initialization of self._predicates_list ?
I think this assumes that get_observation is always called after _get_observation_space and we should not make such assumptions

Reply:
This is necessary since the task initialization logic has not completed yet. But I removed this assumption and fixed this code.